packages/app/src/components/DBRowTable.tsx

```diff
@@ -1267,11 +1268,18 @@ function DBSqlRowTableComponent({
     queryKey: [
       'denoised-rows',
       config,
-      processedRows,
+      denoiseResults,
+      // Only include processed rows if denoising is enabled
+      // This helps prevent the queryKey from getting extremely large
+      // and causing memory issues, when it's not used.
+      ...(denoiseResults ? [processedRows] : []),
```

Comment:
I’m scratching my head as to why this is relevant. even if denoise is disabled, react-query still tries to cache the key that blows up the memory?